java/src/jmri/jmrit/signalling/SignallingSourcePanel.java

```diff
@@ -64,8 +64,8 @@ public SignallingSourcePanel(final SignalMast sourceMast) {
             _signalMastList = sml.getDestinationList();
         }
 
-        InstanceManager.getDefault(LayoutBlockManager.class).addPropertyChangeListener(this);
-        InstanceManager.getDefault(jmri.SignalMastLogicManager.class).addPropertyChangeListener(this);
+        InstanceManager.getDefault(LayoutBlockManager.class).addPropertyChangeListener(SignallingSourcePanel.this);
+        InstanceManager.getDefault(jmri.SignalMastLogicManager.class).addPropertyChangeListener(SignallingSourcePanel.this);
```

Comment:
What's the purpose of this change? It hides the problem without actually fixing it.

The "easy" fix is to move initialization out of the constructor, but it's not always possible and can be risky if the class is used externally (by script or by JMRI as a library like CATS), or if we don't find all the places where the class is created.

Caller:
`Something s = new Something().init();`

Class:
```
public class Something {
    public Something() {
        // Do something
    }

    public Something init() {
        // Do some initialization
       return this;
    }
}
```

An alternative is to have the constructor private and have a static method creating the class. But that makes inheritance of the class impossible. But it's safer.